ec/ecresource/deploymentresource/elasticsearch/v1/schema.go

```diff
@@ -0,0 +1,544 @@
+	"github.com/elastic/terraform-provider-ec/ec/internal/validators"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// These constants are only used to determine whether or not a dedicated
+// tier of masters or ingest (coordinating) nodes are set.
+const (
+	dataTierRolePrefix   = "data_"
+	ingestDataTierRole   = "ingest"
+	masterDataTierRole   = "master"
+	autodetect           = "autodetect"
+	growAndShrink        = "grow_and_shrink"
+	rollingGrowAndShrink = "rolling_grow_and_shrink"
+	rollingAll           = "rolling_all"
+)
+
+// List of update strategies availables.
+var strategiesList = []string{
+	autodetect, growAndShrink, rollingGrowAndShrink, rollingAll,
+}
+
+func ElasticsearchSchema() tfsdk.Attribute {
+	return tfsdk.Attribute{
+		Description: "Required Elasticsearch resource definition",
+		Required:    true,
+		Validators:  []tfsdk.AttributeValidator{listvalidator.SizeAtMost(1)},
+		Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
+			"autoscale": {
+				Type:        types.StringType,
+				Description: `Enable or disable autoscaling. Defaults to the setting coming from the deployment template. Accepted values are "true" or "false".`,
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"ref_id": {
+				Type:        types.StringType,
+				Description: "Optional ref_id to set on the Elasticsearch resource",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					planmodifier.DefaultValue(types.String{Value: "main-elasticsearch"}),
+					resource.UseStateForUnknown(),
+				},
+			},
+			"resource_id": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource unique identifier",
+				Computed:    true,
+				// PlanModifiers: tfsdk.AttributePlanModifiers{
+				// 	resource.UseStateForUnknown(),
+				// },
+			},
+			"region": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource region",
+				Computed:    true,
+				// PlanModifiers: tfsdk.AttributePlanModifiers{
+				// 	resource.UseStateForUnknown(),
+				// },
+			},
+			"cloud_id": {
+				Type:        types.StringType,
+				Description: "The encoded Elasticsearch credentials to use in Beats or Logstash",
+				Computed:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+					resource.RequiresReplaceIf(func(ctx context.Context, state, config attr.Value, path path.Path) (bool, diag.Diagnostics) {
```

Comment:
This should be user specified, why do we need `RequiresReplace*` on this attribute?

Reply:
I don't think plan modifiers in v1 are useful at all - I'm going to delete this.